pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -127,3 +128,80 @@ def close(self) -> None:
         if not self.skip_check_coredump:
             self._check_coredump()
         super().close()
+
+    case_regex = re.compile(r'\((\d+)\)\s\"(.+)\"\s(\[.+\])+')
+    subcase_regex = re.compile(r'\t\((\d+)\)\s\"(.+)\"')
+
+    def get_test_menu(self) -> List:
+        """
+        Get test case list from test menu via UART print.
+
+        This function should be called after `Press ENTER to see the list of tests` prompt is shown.
+
+        Returns:
+            A `list` of avaliable test cases, in `dict`. Each test case's `dict` has following keys:
+                - index: The index of the case, which can be used to run this case.
+                - name: The name of the case.
+                - type: Type of this case, which can be `normal` `multi_stage` or `multi_device`.
+                - keyword: List of additonal keywords of this case. For now, we have `disable` and `ignored`.
+                - group: List of groups of this case, this is usually the component which this case belongs to.
+                - attribute: List of attributes of this case, which is used to describe timeout duration,
+                    test environment, etc.
+                - subcase: List of subcases of this case, if this case is a `multi_stage` or `multi_device` one.
+        """
+        self.write('')
+        menu_block = self.expect("Here's the test menu, pick your combo:(.+)Enter test for running.").group(1)
```

Comment:
Could we make the command to show the menu (here it is a newline) and the prompt message ("Here's the test menu...") configurable, for example as optional arguments to this function?

I have another test app which prints the list of tests after writing the command `test -h` and has a slightly different prompt.